### ISSUE TITLE
Pose Estimation Issues

### ISSUE THREAD

**Issue by chumo-xu** [CUT3R/CUT3R#76]
Hi, thank you for your great work! I'm encountering some issues with pose estimation and point cloud fusion after fine-tuning the model on my dataset, and I hope you could provide some insights!

Here is a summary of my experiments and observations:
1. Using the Official Pre-trained Model (cut3r_512_dpt_4_64.pth)
On Natural Images (e.g., [004](https://github.com/CUT3R/CUT3R/tree/main/examples/004)):Works perfectly. 
Camera poses are correct, depth estimation is accurate, and the static parts of the point cloud fuse well.

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/4128f883-de1f-4032-af06-35d95ca9624e" />

  On Surgical Images (my dataset): Performs poorly.
  Camera pose has errors (though the general trend of motion seems somewhat correct).Depth estimation is inaccurate. And Interestingly, the static parts of the point cloud still seem to show some fusion.

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/4640d415-1913-4062-bf67-2ee6f1d92b63" />

2. After Fine-tuning on Surgical Dataset
I fine-tuned the model using your pre-trained weights(cut3r_512_dpt_4_64.pth), as my dataset is small. And I follow the config/stage3.yaml.
Training & Validation Loss:
All components of the train_loss decreases quickly and converges well.
However, during validation, Regr3DPose and Regr3DPose_ScaleInv remain high and do not converge.
Inference on Natural Images (e.g., [004](https://github.com/CUT3R/CUT3R/tree/main/examples/004)): Performance degrades significantly.
Camera pose is almost static in one case and has minimal movement in another.The point cloud does not fuse.

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/850a46e8-dfb5-40e8-98b1-965c0cf47455" />

Inference on Surgical Images: Mixed results. Camera pose becomes almost completely static. But Depth estimation becomes very good. The point cloud does not fuse at all.

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/5f8277b0-315c-4b60-bbae-d6a4bdebbab9" />

My Analysis and Hypothesis
I believe the issue might stem from the nature of my fine-tuning data and training process.
1. The camera motion in my surgical dataset is often very small between adjacent frames.
2. I did not shuffle the dataset, so the model was trained on sequences of 4 views with minimal-to-no displacement.
This might have led the model to fall into a local minimum where it assumes the camera pose is always static. This hypothesis could explain:
- Good Single-View Depth: With a static pose assumption, the model focuses on optimizing single-view depth estimation, leading to a low self-view 3D loss.
- Non-fusing Point Cloud: Since the model predicts no ego-motion, it cannot correctly warp points from one frame to another, hence the lack of fusion.
- But I don't get why the Validation 3D Loss is high because the displacement in validation dataset is also small.

Questions
1.Does my analysis seem plausible? Is this a potential issue when fine-tuning on datasets with small inter-frame motion?
2.I saw in another issue #68  that you mentioned training might not require ground truth camera poses. Would you recommend trying to train without GT poses in my case to force the model to learn relative motion?

My data preprocessing and loading follow the format of the Point Odyssey dataset.

My apologies for such a long and detailed issue; I wanted to provide as much context as possible. Any advice would be greatly appreciated. Thank you very much!

**Comment by Forrest-110** [comment]
Hi @chumo-xu! Thank you for your inquiry!

Have you checked whether the distributions of your training and validation sets are similar? It's unusual to observe such a large gap in performance, especially regarding camera poses. Also, have you looked at this [issue](https://github.com/CUT3R/CUT3R/issues/53) to ensure that the camera supervision is applied correctly?

If everything seems correct so far, I’d recommend inspecting the scale of the pose when computing the loss. You can refer to these [lines](https://github.com/CUT3R/CUT3R/blob/b9c07dda7ee8737fac20e7d5b4dda4f4b0ca7b68/src/dust3r/losses.py#L652-L657) to see how poses are normalized—this can help diagnose scale-related issues.

If the pose scale appears appropriate, you might consider supervising relative poses instead. Note that this [issue](https://github.com/CUT3R/CUT3R/issues/68) pertains to inference rather than training, so it may not be directly relevant. One possible direction is to add loss terms on pairwise poses. Additionally, you could experiment with computing the scale of the pointmap and camera poses separately, as seen [here](https://github.com/CUT3R/CUT3R/blob/b9c07dda7ee8737fac20e7d5b4dda4f4b0ca7b68/src/dust3r/losses.py#L624).

Let me know if any of this helps!

**Comment by chumo-xu** [comment]
Thank you so much for your prompt reply!
I find my data loading code is based on the Point Odyssey configuration, where the sampling probability ratio is set to [0.9, 0.05, 0.05].
(            img_mask, ray_mask = self.get_img_and_ray_masks(
                self.is_metric, v, rng, p=[0.9, 0.05, 0.05]                                                  )
This means only 10% of the rays are used for pose and image information. Could this low percentage be the reason for the inaccurate pose estimation?
And Would it be better for pose estimation if I change the ratio to [0.6, 0.3, 0.1]?
BTW I have one more question: Why not set it to [0, 0, 1] to use all the pose and image data for training?

Thanks again for your help!

**Comment by Forrest-110** [comment]
Hi @chumo-xu, apologies for the delayed response!

Let me clarify the meaning of the ratio [0.9, 0.05, 0.05], which corresponds to the distribution of input modes: image-only vs. raymap-only vs. both.

- 90% of the data uses image-only input, which reflects our main task—reconstructing 3D structures from unposed images.

- 5% uses raymap-only input, supporting our auxiliary task—inferring unseen structures via raymaps.

- 5% includes both image and pose inputs. This is intended to make the model compatible with scenarios where ground-truth pose is available, though this is not the primary objective.

Hope this helps clarify things!

**Comment by chumo-xu** [comment]
Thanks for your reply!
Given that my dataset's depth information is relatively accurate but the camera poses are not, and I don't need to infer unknown regions, can I set the ratio to 100% for image-only input? This would mean not using the camera poses at all in my training.

**Comment by Forrest-110** [comment]
I think that would be fine. In that case, ground-truth camera poses are only used as supervision signals for the predicted pose output.

**Comment by chumo-xu** [comment]
Thanks for the clarification.
Just to be absolutely sure, does this mean I still need to provide a ground-truth pose, even if it's not perfectly accurate, for the model to use as a supervision signal for its own pose prediction?
In other words, the model isn't using my provided poses for the 3D reconstruction task, but it does need them for the separate task of learning how to predict poses. Is it not possible to do a truly pose-free training in this scenario?

**Comment by Forrest-110** [comment]
I'm afraid it's not feasible to train entirely without camera poses. 3D reconstruction involves recovering structure within a certain world coordinate system, which requires ground-truth camera information—either explicitly (through camera poses) or implicitly (via cross-view point maps).

If your poses are not perfectly accurate, I would recommend first verifying the ground truth in 3D space by visualizing the point clouds in the world frame to check for alignment. If they appear well-aligned, then the inaccurate poses should be fine to use. Otherwise, you might be looking for single-frame depth estimation methods instead.

**Comment by chumo-xu** [comment]
Thanks! I’ll check it.

**Comment by fangli333** [comment]
> Thanks! I’ll check it.

Hi,

have you figure out why the camera pose is amost static? Thank you!
